### DISCUSSION TITLE
How to load changes from pivot table

### DISCUSSION THREAD

**Discussion by takeshiyu** [spatie/laravel-activitylog#933] [Q&A]
I have a one-to-many relationship model: an order have an infinite number of details, and the definition is like following:

Order.php:
```php
class Order extends Model
{
    use \App\Models\Concerns\Auditable;

    public function details()
    {
        return $this->hasMany(Detail::class);
    }
}
```
Detail.php: 
```php
class Detail extends Pivot
{
    use \App\Models\Concerns\Auditable;

    public $incrementing = true;
    protected $touches = ['order'];

    public function order()
    {
        return $this->belongsTo(Order::class);
    }
}
```

And I use the following code to update the `Order` and `Detail`:

```php
$order = $this->updateOrder($order, $input);

foreach ($input['details'] as $detail) {
    $pivot = $order->details()->where('id', $detail['id'])->first();
    $pivot->quantity = $detail['quantity'];
    $order->details()->save($pivot);
}
```

When changes are only on `Detail` ( `Order` is no need to update ), `laravel-activitylog` will only record the changes of `Detail` in database. And how do I show all the changes of `order` including the details ?

**Comment by Gummibeer** [comment]
Hey,

first thing that comes to mind are the Log Batches we added in v4: https://spatie.be/docs/laravel-activitylog/v4/advanced-usage/batch-logs

With these you can group a batch of logs and retrieve all logs to one batch afterward.

This would be the most reliable way to do it as everything is done in the logs table and doesn't rely on any other table values.
